File: pdu/python/geometry_msgs/pdu_cdr_conv_TransformStamped.py
from .pdu_pytype_TransformStamped import TransformStamped
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..std_msgs.pdu_cdr_conv_Header import *
from ..geometry_msgs.pdu_cdr_conv_Quaternion import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *
from ..geometry_msgs.pdu_cdr_conv_Transform import *
from ..geometry_msgs.pdu_cdr_conv_Vector3 import *



def py_to_cdr_body_TransformStamped(writer: CdrWriter, src: TransformStamped):
    py_to_cdr_body_Header(writer, src.header)
    writer.write_string(src.child_frame_id)
    py_to_cdr_body_Transform(writer, src.transform)


def cdr_body_to_py_TransformStamped(reader: CdrReader, dst: TransformStamped):
    cdr_body_to_py_Header(reader, dst.header)
    dst.child_frame_id = reader.read_string()
    cdr_body_to_py_Transform(reader, dst.transform)
    return dst


def py_to_cdr_TransformStamped(src: TransformStamped) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_TransformStamped(writer, src)
    return writer.bytes()


def cdr_to_py_TransformStamped(cdr_payload) -> TransformStamped:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = TransformStamped()
    return cdr_body_to_py_TransformStamped(reader, dst)

File: pdu/javascript/geometry_msgs/pdu_cdr_conv_TransformStamped.js
import { TransformStamped } from './pdu_jstype_TransformStamped.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { Quaternion } from '../geometry_msgs/pdu_jstype_Quaternion.js';
import { PduQuaternionConverter } from '../geometry_msgs/pdu_cdr_conv_Quaternion.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';
import { Transform } from '../geometry_msgs/pdu_jstype_Transform.js';
import { PduTransformConverter } from '../geometry_msgs/pdu_cdr_conv_Transform.js';
import { Vector3 } from '../geometry_msgs/pdu_jstype_Vector3.js';
import { PduVector3Converter } from '../geometry_msgs/pdu_cdr_conv_Vector3.js';


export class PduTransformStampedConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { TransformStamped } src
     */
    static to_cdr_body(writer, src) {
        PduHeaderConverter.to_cdr_body(writer, src.header);
        writer.write_string(src.child_frame_id);
        PduTransformConverter.to_cdr_body(writer, src.transform);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { TransformStamped } dst
     * @returns { TransformStamped }
     */
    static cdr_body_to_js(reader, dst) {
        PduHeaderConverter.cdr_body_to_js(reader, dst.header);
        dst.child_frame_id = reader.read_string();
        PduTransformConverter.cdr_body_to_js(reader, dst.transform);
        return dst;
    }

    /**
     * @param { TransformStamped } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { TransformStamped }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new TransformStamped());
    }
}
